syntax = "proto3";

package tunnel_api;


enum ChannelType {
    REQ_RES = 0;
    PUBLISH_SUBSCRIBE = 1; 
}

enum Service {
    STOCK = 0;
    STOCK_ORDER = 1;
}

enum Role {
    CLIENT = 0;
    PROVIDER = 1;
}

message Node {
    string vendor = 1;
    ChannelType channel_type = 2;
    Role role = 3;
    string name = 4;
    Service service = 5;
}

message Result {
    enum ResultType {
        OK = 0;
        UNKNOWN_ERROR = 1;
        NOT_AVAILABLE = 2;
        MESSAGE_ERROR = 3;
    }

    ResultType result_code = 1;
}
